src/main/java/org/folio/circulation/resources/NoticeCirculationRulesEngineResource.java

```diff
@@ -23,7 +25,8 @@ public NoticeCirculationRulesEngineResource(String applyPath, String applyAllPat
   }
 
   @Override
-  protected CompletableFuture<Result<String>> getPolicyId(MultiMap params, Drools drools, Location location) {
+  protected CompletableFuture<Result<Pair<String, List<String>>>> getPolicyIdAndRuleMatch(
+    MultiMap params, Drools drools, Location location) {
     return CompletableFuture.completedFuture(succeeded(drools.noticePolicy(params, location)));
```

Comment:
Personally, I would statically import the `completedFuture` method to shorten the length of this line and improve comprehension

```suggestion
    return completedFuture(succeeded(drools.noticePolicy(params, location)));
```